src/validation.cpp

```diff
@@ -1166,8 +1166,7 @@ static std::pair<CAmount, CAmount> GetBlockSubsidyHelper(int nPrevBits, int nPre
         nSubsidy -= nSubsidy/14;
     }
 
-    // this is only active on devnets
-    if (nPrevHeight < consensusParams.nHighSubsidyBlocks) {
+    if (isDevnet && nPrevHeight < consensusParams.nHighSubsidyBlocks) {
```

Comment:
couldn't high subsidy be useful on regtest?

Reply:
why do we need a check here `isDevnet` ?

```
    int nHighSubsidyBlocks{0};
```
it seem as it can be not-zero only for non-devnet. can remove this condition then:

```suggestion
    // this is only active on devnets
    if (nPrevHeight < consensusParams.nHighSubsidyBlocks) {
```

Reply:
Nothing enforces "`nHighSubsidyBlocks == 0` on non-devnets" so I'd prefer having it in code rather that simply in comments

Reply:
we could do
```suggestion
    if (nPrevHeight < consensusParams.nHighSubsidyBlocks) {
        assert(isDevnet);
```
instead if you don't like it inside `if` clause

Reply:
let's add better `assert(!isDevnet && consensusParams.nHightSsubisdyBlocks != 0)` to **enforce** it?